tools/bootstrap_commitlogs/main/main.go

```diff
@@ -0,0 +1,233 @@
+package main
+
+import (
+	"flag"
+	"io"
+	"net/http"
+	_ "net/http/pprof"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/m3db/m3db/encoding"
+	"github.com/m3db/m3db/encoding/m3tsz"
+	"github.com/m3db/m3db/persist/fs"
+	"github.com/m3db/m3db/persist/fs/commitlog"
+	"github.com/m3db/m3db/retention"
+	"github.com/m3db/m3db/storage/block"
+	"github.com/m3db/m3db/storage/bootstrap"
+	"github.com/m3db/m3db/storage/bootstrap/bootstrapper"
+	commitlogsrc "github.com/m3db/m3db/storage/bootstrap/bootstrapper/commitlog"
+	"github.com/m3db/m3db/storage/bootstrap/result"
+	"github.com/m3db/m3db/storage/namespace"
+	"github.com/m3db/m3db/ts"
+	"github.com/m3db/m3db/x/io"
+	"github.com/m3db/m3x/instrument"
+	xlog "github.com/m3db/m3x/log"
+	xtime "github.com/m3db/m3x/time"
+)
+
+var (
+	pathPrefixArg           = flag.String("path-prefix", "/var/lib/m3db", "Path prefix - must contain a folder called 'commitlogs'")
+	namespaceArg            = flag.String("namespace", "metrics", "Namespace")
+	blockSizeArg            = flag.Duration("block-size", 10*time.Minute, "Block size")
+	flushSizeArg            = flag.Int("flush-size", 524288, "Flush size of commit log")
+	bootstrapRetentionArg   = flag.Duration("retention", 48*time.Hour, "Retention")
+	shardsCountArg          = flag.Int("shards-count", 8192, "Shards count - set number too bootstrap all shards in range")
+	shardsArg               = flag.String("shards", "", "Shards - set comma separated list of shards")
+	debugListenAddressArg   = flag.String("debug-listen-address", "", "Debug listen address - if set will expose pprof, i.e. ':8080'")
+	currentUnixTimestampArg = flag.String("current-unix-timestamp", currentUnixTimestampString(), "Current unix timestamp - If set will perform the bootstrap as if this was the current time, defaults to current time")
```

Comment:
Any reason to not just make the Var a int64 instead of a string?

Reply:
good idea will fix